crates/api_common/src/request.rs

```diff
@@ -295,32 +302,31 @@ async fn generate_pictrs_thumbnail(image_url: &Url, context: &LemmyContext) -> L
     encode(image_url.as_str())
   );
 
-  let response = context
+  let res: PictrsResponse = context
     .client()
     .get(&fetch_url)
     .timeout(REQWEST_TIMEOUT)
     .send()
+    .await?
+    .json()
     .await?;
 
-  let response: PictrsResponse = response.json().await?;
-
-  if response.msg == "ok" {
-    let thumbnail_url = Url::parse(&format!(
-      "{}/pictrs/image/{}",
-      context.settings().get_protocol_and_hostname(),
-      response.files.first().expect("missing pictrs file").file
-    ))?;
-    for uploaded_image in response.files {
-      let form = LocalImageForm {
-        local_user_id: None,
-        pictrs_alias: uploaded_image.file.to_string(),
-        pictrs_delete_token: uploaded_image.delete_token.to_string(),
-      };
-      LocalImage::create(&mut context.pool(), &form).await?;
-    }
+  if let Some(image) = res.files.unwrap_or_default().first() {
```

Comment:
Use this instead:

```
let image = res.files
  .unwrap_or_default()
  .first()
  .ok_or(lemmyerrortype::PictrsResponseError(res.msg))?;
```

Reply:
Thx. I had to separate it into two vars in order to make rusts borrow checker happy, but its still cleaner.